src/utilities/schemaComparisons.js

```diff
@@ -0,0 +1,220 @@
+} from '../type/definition';
+import {
+  GraphQLSchema,
+} from '../type/schema';
+import type {
+  TypeMap,
+} from '../type/schema';
+
+/**
+ * Given two schemas, returns a Set containing descriptions of all the types of
+ * breaking changes covered by the other functions down below.
+ */
+export function findBreakingChanges(
+  oldSchema: GraphQLSchema,
+  newSchema: GraphQLSchema
+): Set<string> {
+  return new Set([
+    ...Array.from(findRemovedTypes(oldSchema, newSchema)),
+    ...Array.from(findTypesThatChangedType(oldSchema, newSchema)),
+    ...Array.from(findBreakingFieldChanges(oldSchema, newSchema)),
+    ...Array.from(findTypesRemovedFromUnions(oldSchema, newSchema)),
+    ...Array.from(findValuesRemovedFromEnums(oldSchema, newSchema))
+  ]);
+}
+
+/**
+* Given two schemas, returns a Set containing descriptions of any breaking
+* changes in the newSchema related to removing an entire type.
+*/
+export function findRemovedTypes(
+  oldSchema: GraphQLSchema,
+  newSchema: GraphQLSchema
+): Set<string> {
+  const oldTypes: Set<string> = new Set(Object.keys(oldSchema.getTypeMap()));
+  const newTypes: Set<string> = new Set(Object.keys(newSchema.getTypeMap()));
+  const removedTypes: Set<string> = new Set(
+    Array.from(oldTypes).filter(typeName => !newTypes.has(typeName)).map(
+     type => `${type} was removed`
+    )
+  );
+  return removedTypes;
+}
+
+/**
+* Given two schemas, returns a Set containing descriptions of any breaking
+* changes in the newSchema related to changing the type of a type.
+*/
+export function findTypesThatChangedType(
+ oldSchema: GraphQLSchema,
+ newSchema: GraphQLSchema
+): Set<string> {
+  const oldTypeMap: TypeMap = oldSchema.getTypeMap();
+  const newTypeMap: TypeMap = newSchema.getTypeMap();
+
+  const typesThatChangedType: Array<string> = [];
+  for (const typeName: string in oldTypeMap) {
+    if (!newTypeMap[typeName]) {
+      continue;
+    }
+    const oldType: GraphQLType = oldTypeMap[typeName];
+    const newType: GraphQLType = newTypeMap[typeName];
+    if (!(oldType instanceof newType.constructor)) {
+      typesThatChangedType.push(`${typeName} changed from a ` +
+        `${oldType.constructor.name} to a ${newType.constructor.name}`);
+    }
+  }
+  return new Set(typesThatChangedType);
+}
+
+/**
+* Given two schemas, returns a Set containing descriptions of any breaking
+* changes in the newSchema related to the fields on a type. This includes if
+* a field has been removed from a type or if a field has changed type.
+*/
+export function findBreakingFieldChanges(
+  oldSchema: GraphQLSchema,
+  newSchema: GraphQLSchema
+): Set<string> {
+  const oldTypeMap: TypeMap = oldSchema.getTypeMap();
+  const newTypeMap: TypeMap = newSchema.getTypeMap();
+
+  const breakingFieldChanges: Array<string> = [];
+  for (const typeName: string in oldTypeMap) {
+    if (!{}.hasOwnProperty.call(oldTypeMap,typeName)) {
+      continue;
+    }
+    const oldType:GraphQLType = oldTypeMap[typeName];
```

Comment:
nit: space after `:`

But also, you don't need to add types to variables you create - Flow infers all of that for you. Only need to annotate function args & returns. You can remove all of them and Flow will keep on chugging.